packages/TorneloScoresheet/src/hooks/appMode/editMoveState.ts

```diff
@@ -118,8 +126,21 @@ const rebuildHistory = (
         move.promotion,
         positionOccurances,
       );
+
+      // return a skip if move impossible
       if (!nextMoveResult) {
-        return null;
+        previousPly = {
+          moveNo: move.moveNo,
+          startingFen: previousStartingFen,
+          drawOffer: move.drawOffer,
+          player: move.player,
+          type: PlyTypes.SkipPly,
+          legality: checkMoveLegality(previousStartingFen, positionOccurances),
```

Comment:
The legality of this move needs to be checked with the fen resulting from this move, ie you need `const nextFen = chessEngine.skipTurn(previousStartingFen)` before declaring previousPly, then you need to call checkMoveLegality with this next fen,
 I just noticed there is a bug in the previous version, in line 138, it should be checking the legality with the next fen too!